### PR TITLE
[WebNN] Support SimplifiedLayerNormalization op

### PR DESCRIPTION
**Pull request by Honry**
WebNN doesn't provide dedicate op for SimplifiedLayerNormalization, use a couple of WebNN ops to emulate it in WebNN EP.

X --> Pow --> ReduceMean --> Add --> Sqrt --> Div -> Mul

### CODE REVIEW

**Comment by Honry** [comment]
@fdwr, @guschmue, PTAL, thanks!

File: onnxruntime/core/providers/webnn/builders/impl/normalization_op_builder.cc [hunk 1]
```diff
@@ -84,14 +85,71 @@ Status NormalizationOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder
...
     std::iota(axes.begin(), axes.end(), axis);
 
-    options.set("axes", emscripten::val::array(axes));
-    output = model_builder.GetBuilder().call<emscripten::val>("layerNormalization", input, options);
+    if (op_type == "LayerNormalization") {
+      options.set("axes", emscripten::val::array(axes));
+      output = model_builder.GetBuilder().call<emscripten::val>("layerNormalization", input, options);
+    } else {  // SimplifiedLayerNormalization
+      /**
+      WebNN doesn't support SimplifiedLayerNormalization, decompose it into a series of ops as follows:
```
Comment (fdwr) [inline onnxruntime/core/providers/webnn/builders/impl/normalization_op_builder.cc:99]:
```suggestion
      WebNN doesn't support SimplifiedLayerNormalization. So decompose it into a series of ops:
```

File: onnxruntime/core/providers/webnn/builders/impl/normalization_op_builder.cc [hunk 2]
```diff
@@ -84,14 +85,71 @@ Status NormalizationOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder
...
+      // Sqrt
+      common_options.set("label", node.Name() + "_sqrt");
+      emscripten::val sqrt = model_builder.GetBuilder().call<emscripten::val>("sqrt", add, common_options);
+
+      // Div
+      common_options.set("label", node.Name() + "_div");
+      emscripten::val div = model_builder.GetBuilder().call<emscripten::val>("div", input, sqrt, common_options);
+
+      // Mul
+      common_options.set("label", node.Name() + "_div");
```
Comment (fdwr) [inline onnxruntime/core/providers/webnn/builders/impl/normalization_op_builder.cc:150]:
```suggestion
      common_options.set("label", node.Name() + "_mul");
```

**Review by fdwr — COMMENTED** [review]
It looks correct (ignoring minor things), and it's okay to support a few com.microsoft-specific things for convenience (though, pure ONNX models are cleaner), but where is the `cast` for higher precision with float16 weights? I thought that was responsible for the poor results on Phi-3/

**Comment by Honry** [comment]
> where is the `cast` for higher precision with float16 weights? I thought that was responsible for the poor results on Phi-3, but if these are just equivalent to decomposed ONNX ops (as if they were already decomposed in the model to begin with), then what's the cause?

The Cast was inserted in the model file (Cast -> SimplifiedLayerNormalization -> Cast), I don't want to introduce more workaround in the EP.

If a model contains `SimplifiedLayerNormalization` node, this patch would help decompose it into several WebNN ops rather then fallback to CPU.

File: onnxruntime/core/providers/webnn/builders/impl/normalization_op_builder.cc [hunk 3]
```diff
@@ -84,14 +85,71 @@ Status NormalizationOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder
...
+      /**
+      WebNN doesn't support SimplifiedLayerNormalization, decompose it into a series of ops as follows:
+      X --> Pow --> ReduceMean --> Add --> Sqrt --> Div -> Mul
+            ^          ^           ^                ^      ^
+            |          |           |                |      |
+            Y:2        axis     B:epsilon           A:X  A:scale
+      */
+
+      int32_t input_type;
+      ORT_RETURN_IF_NOT(GetType(*input_defs[0], input_type, logger), "cannot get input type");
```
[the remark this answers is not in the document]
Comment (Honry) [reply onnxruntime/core/providers/webnn/builders/impl/normalization_op_builder.cc:107]: Should be uppercase. :)

**Comment by Honry** [comment]
@fdwr, comments addressed, pls. take another look, thank you!

**Review by fdwr — DISMISSED** [review]
👍

**Comment by fdwr** [comment]
Sorry, merge conflicts :(.

**Comment by Honry** [comment]
> Sorry, merge conflicts :(.

NP, fixed.

**Review by fdwr — APPROVED** [review]

**Pull request merged**